rust/parser/src/lexer.rs

```diff
@@ -267,55 +267,60 @@ impl<'alloc> Lexer<'alloc> {
             self.chars.next();
             builder.push_matching(ch);
         }
+        let has_different = builder.has_different();
         let text = builder.finish(&self);
 
-        let id = match &text as &str {
-            "as" => TerminalId::As,
-            "async" => TerminalId::Async,
-            "await" => TerminalId::Await,
-            "break" => TerminalId::Break,
-            "case" => TerminalId::Case,
-            "catch" => TerminalId::Catch,
-            "class" => TerminalId::Class,
-            "const" => TerminalId::Const,
-            "continue" => TerminalId::Continue,
-            "debugger" => TerminalId::Debugger,
-            "default" => TerminalId::Default,
-            "delete" => TerminalId::Delete,
-            "do" => TerminalId::Do,
-            "else" => TerminalId::Else,
-            "export" => TerminalId::Export,
-            "extends" => TerminalId::Extends,
-            "finally" => TerminalId::Finally,
-            "for" => TerminalId::For,
-            "from" => TerminalId::From,
-            "function" => TerminalId::Function,
-            "get" => TerminalId::Get,
-            "if" => TerminalId::If,
-            "import" => TerminalId::Import,
-            "in" => TerminalId::In,
-            "instanceof" => TerminalId::Instanceof,
-            "let" => TerminalId::Let,
-            "new" => TerminalId::New,
-            "of" => TerminalId::Of,
-            "return" => TerminalId::Return,
-            "set" => TerminalId::Set,
-            "static" => TerminalId::Static,
-            "super" => TerminalId::Super,
-            "switch" => TerminalId::Switch,
-            "target" => TerminalId::Target,
-            "this" => TerminalId::This,
-            "throw" => TerminalId::Throw,
-            "try" => TerminalId::Try,
-            "typeof" => TerminalId::Typeof,
-            "var" => TerminalId::Var,
-            "void" => TerminalId::Void,
-            "while" => TerminalId::While,
-            "with" => TerminalId::With,
-            "yield" => TerminalId::Yield,
-            "null" => TerminalId::NullLiteral,
-            "true" | "false" => TerminalId::BooleanLiteral,
-            _ => TerminalId::Name,
+        let id = if has_different {
```

Comment:
I think it might be a good idea to add a comment here about what this is doing. Maybe something like:
"if matching literal sequence contains an escape sequence, the TerminalId should always be `Name`, see issue #93" -- or something to this effect?